lib/msal-core/samples/react-sample-app/src/AuthProvider.js

```diff
@@ -0,0 +1,173 @@
+import React, { Component } from "react";
+import * as Msal from "msal";
+
+const loginRequest = {
+    scopes: ["openid", "profile", "User.Read"]
+};
+
+const tokenRequest = {
+    scopes: ["Mail.Read"]
+};
+
+const graphConfig = {
+    graphMeEndpoint: "https://graph.microsoft.com/v1.0/me",
+    graphMailEndpoint: "https://graph.microsoft.com/v1.0/me/messages"
+};
+
+const msalApp = new Msal.UserAgentApplication({
+    auth: {
+        clientId: "245e9392-c666-4d51-8f8a-bfd9e55b2456",
+        authority: "https://login.microsoftonline.com/common",
+        validateAuthority: true,
+        postLogoutRedirectUri: "http://localhost:3000"
+    },
+    cache: {
+        cacheLocation: "localStorage",
+        storeAuthStateInCookie: true
+    }
+});
+
+const callMSGraph = async (url, accessToken) => {
+    const response = await fetch(url, {
+        headers: {
+            Authorization: `Bearer ${accessToken}`
+        }
+    });
+
+    return response.json();
+};
+
+const requiresInteraction = errorMessage => {
+    if (!errorMessage || !errorMessage.length) {
+        return false;
+    }
+
+    return (
+        errorMessage.indexOf("consent_required") !== -1 ||
+        errorMessage.indexOf("interaction_required") !== -1 ||
+        errorMessage.indexOf("login_required") !== -1
+    );
+};
+
+const isIE = () => {
+    const ua = window.navigator.userAgent;
+    const msie = ua.indexOf("MSIE ");
+    const msie11 = ua.indexOf("Trident/");
+
+    return msie > 0 || msie11 > 0;
+};
+
+export default C =>
+    class AuthProvider extends Component {
```

Comment:
Thoughts on adding a prop that is `autoLogin`? 

maybe not because you would need an error page if they don't authenticate?

Reply:
Yeah I think that is something we could add incrementally.